src/formatter/index.ts

```diff
@@ -238,4 +239,47 @@ export class QueryFormatter {
     }
     return query;
   }
+
+  isSetStatement(query: string): boolean {
+    return query.trim().toLowerCase().startsWith(SET_PREFIX);
+  }
+
+  trimTrailingSemicolon(text: string): string {
+    let end = text.length;
+    while (end > 0 && text[end - 1] == ";") {
+      end--;
+    }
+    return text.substring(0, end);
+  }
+
+  trimStringQuotes(text: string): string {
+    if (text.length < 2) {
+      return text;
+    }
+
+    const firstChar = text[0];
+    const lastChar = text[text.length - 1];
+    if (firstChar === lastChar && (firstChar === "'" || firstChar === '"')) {
+      return text.substring(1, text.length - 1);
+    }
+
+    return text;
+  }
+
+  splitSetStatement(query: string): [string, string] {
+    query = this.trimTrailingSemicolon(query);
+
+    const strippedEquation = query.trim().substring(SET_PREFIX.length).trim();
+    const items = strippedEquation.split(/=(.*)/s);
+    // split with a capturing group for some reason always returns
+    // an empty string as the third element, we can just ignore it
+    if (items.length !== 3) {
+      throw new Error("Invalid SET statement format");
+    }
+    const [key, value] = items.slice(0, 2);
```

Comment:
I feel like having capture groups here and regex is a bit overkill. One simple solution I came across is to find the index:
```
const index = strippedEquation.indexOf("=");
const key = strippedEquation.slice(0, index);
const value = strippedEquation.slice(index + 1);
```
IMO, this is safer and very straight forward.